### PR TITLE
Add travis support.

### PR DESCRIPTION
**Pull request by Yashwants19**
Hi @rcurtin, I have tried to add travis-ci support to this repository. Please take a look.

Thank You.

### CODE REVIEW

File: rel/deployment.md [hunk 1]
```diff
@@ -0,0 +1,15 @@
...
+ 1. Check out mlpack code.
+ 2. Configure and build the `go` target: `make go`
+ 3. Copy the complete folder of the Go bindings (`build/src/mlpack/bindings/go/src/mlpack.org/v1/mlpack`)
+    to the root of mlpack-go repository.
+ 4. Remove all the `.so` files.
+```sh
+rm -f *.so *.so.*
+```
+ 5. Manually change the `PACKAGE_VERSION` in `MAKEFILE`.
+ 6. Commit any changed files and any added files in `root` and `capi/` folder of mlpack-go repository.
```
Comment (rcurtin) [inline rel/deployment.md:15]: Awesome, thank you for adding this!

File: Makefile [hunk 2]
```diff
@@ -47,7 +54,16 @@ deps_fedora:
...
-endif
+
+# Download and install Armadillo.
+setup_armadillo:
+	rm -rf $(TMP_DIR)armadillo
+	mkdir $(TMP_DIR)armadillo
+	cd $(TMP_DIR)armadillo
+	curl https://ftp.fau.de/macports/distfiles/armadillo/armadillo-$(ARMA_VERSION).tar.xz \
+    	| tar -xvJ &&  cd armadillo* && \
+    	cmake . && make && sudo make install && cd ..  && rm -rf armadillo*
```
Comment (rcurtin) [inline Makefile:65]: Nice, great to have this extra support, although honestly I think it's okay to force the user to install Armadillo themselves.

**Review by rcurtin — COMMENTED** [review]
This looks good, thanks @Yashwants19!  I just hooked up Travis support, so let's see if it builds...

File: Makefile (hunk 2 above)
Comment (Yashwants19) [reply Makefile:65]: Yeah, we can remove this, then we must also add this information in quick start or some other place that user is only responsible for installing armadillo.

Comment (rcurtin) [reply Makefile:65]: Up to you; it looks like now the Makefile will automatically install all dependencies *except* Armadillo?  I don't think I fully understood the workflow before.  I didn't realize that the Makefile would use apt or brew or yum/dnf when possible, and only build Armadillo from source if that wasn't available.  Anyway, I don't see any issue re-adding it.

Let me know what you think, and then we can go ahead and merge this. :+1:

Comment (Yashwants19) [reply Makefile:65]: >  Anyway, I don't see any issue re-adding it.

This sounds good, We can re-add armadillo in the `MAKEFILE`. :+1:

Comment (Yashwants19) [reply Makefile:65]: Should we add proper documentation regarding this workflow in `go-quickstart`?

**Comment by Yashwants19** [comment]
Hi @rcurtin should we merge this or am I missing something here?

File: Makefile (hunk 2 above)
Comment (rcurtin) [reply Makefile:65]: Yeah, if you want to update the Go quickstart, that would be great too. :+1: Thanks!

**Comment by rcurtin** [comment]
Sorry, just hadn't had a chance to get back to it.  Everything looks great! :100:

**Comment by Yashwants19** [comment]
Thanks @rcurtin for merging this.

**Pull request merged**